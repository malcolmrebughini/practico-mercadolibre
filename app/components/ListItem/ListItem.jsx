import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import freeShipping from './assets/ic_shipping.png';
import styles from './styles.scss';


function ListItem(props) {
  const linkTo = {
    pathname: `/items/${props.id}`,
  };

  return (
    <li className={styles.item}>
      <Link title={props.title} to={linkTo}>
        <div className={styles.thumbnail}>
          <img src={props.thumbnail} alt={props.title} />
        </div>
        <div className={styles.description}>
          <span className={styles.price}>
            $ {props.price}
            {
              props.freeShipping &&
                <img alt="envio gratis" className={styles.freeShipping} src={freeShipping} />
            }
          </span>
          <span className={styles.title}>{props.title}</span>
        </div>
        <div className={styles.location}>
          {props.location}
        </div>
      </Link>
    </li>
  );
}


ListItem.propTypes = {
  id: PropTypes.string,
  thumbnail: PropTypes.string,
  title: PropTypes.string,
  price: PropTypes.number,
  location: PropTypes.string,
  freeShipping: PropTypes.bool,
};


export default ListItem;
