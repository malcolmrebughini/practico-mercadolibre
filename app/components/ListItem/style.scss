.item {
  padding: 16px;
  border-top: 1px solid rgba(51, 51, 51, 0.1);

  &:first-of-type {
    border-top: none;
  }

  a {
    text-decoration: none;
    display: flex;
    flex-direction: row;
    color: #333;

    &:visited {
      color: #333;
    }
  }

  .thumbnail {
    img {
      width: 180px;
      height: 180px;
      border-radius: 4px
    }
  }

  .description {
    padding-left: 16px;
    flex-grow: 2;

    .title {
      display: block;
      font-weight: 300;
      font-size: 14px;
      margin-top: 32px;

      @media (min-width: 768px) {
        font-size: 18;
        width: 300px;
      }
    }

    .price {
      display: block;
      font-size: 24px;
    }

    .freeShipping {
      margin-left: 10px;
    }
  }

  .location {
    display: none;

    @media (min-width: 768px) {
      display: block;
      flex-grow: 1;
      font-size: 12px;
      flex-basis: 100px;
    }
  }
}