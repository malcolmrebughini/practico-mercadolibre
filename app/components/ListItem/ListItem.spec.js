/* eslint-env jasmine */
/* eslint react/jsx-filename-extension: 0 */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ListItem from './ListItem';


describe('ListItem', () => {
  it('should render correctly', () => {
    const wrapper = shallow(
      <ListItem
        thumbnail="http://someurl.com/someimage.jpg"
        price={200}
        freeShipping
        title="Some title"
        location="San Nicolas"
      />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
