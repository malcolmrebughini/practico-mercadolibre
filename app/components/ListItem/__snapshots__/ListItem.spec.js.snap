// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render correctly 1`] = `
<li
  className="item"
  itemScope={true}
  itemType="http://schema.org/Product"
>
  <Link
    replace={false}
    title="Some title"
    to={
      Object {
        "pathname": "/items/undefined",
      }
    }
  >
    <div
      className="thumbnail"
    >
      <img
        alt="Some title"
        itemProp="image"
        src="http://someurl.com/someimage.jpg"
      />
    </div>
    <div
      className="description"
    >
      <span
        className="price"
      >
        $ 
        200
        <img
          alt="envio gratis"
          className="freeShipping"
          src="some-static-file"
        />
      </span>
      <span
        className="title"
        itemProp="name"
      >
        Some title
      </span>
    </div>
    <div
      className="location"
    >
      San Nicolas
    </div>
  </Link>
</li>
`;
